Sources_v3/StreamChat/WebSocketClient/EventMiddlewares/MessageReactionsMiddleware.swift

```diff
@@ -0,0 +1,38 @@
+//
+// Copyright © 2020 Stream.io Inc. All rights reserved.
+//
+
+import Foundation
+
+struct MessageReactionsMiddleware<ExtraData: ExtraDataTypes>: EventMiddleware {
+    let database: DatabaseContainer
+    
+    func handle(event: Event, completion: @escaping (Event?) -> Void) {
+        guard let reactionEvent = event as? EventWithReactionPayload else {
+            completion(event)
+            return
+        }
+        
+        database.write({ session in
+            if let event = reactionEvent as? ReactionNewEvent<ExtraData> {
+                try session.saveReaction(payload: event.reaction)
+            } else if let event = reactionEvent as? ReactionUpdatedEvent<ExtraData> {
+                try session.saveReaction(payload: event.reaction)
+            } else if let event = reactionEvent as? ReactionDeletedEvent<ExtraData> {
+                let dto = try session.reaction(
+                    messageId: event.messageId,
+                    userId: event.userId,
+                    type: event.reactionType
+                )
+                dto.flatMap(session.delete)
```

Comment:
??

Reply:
Ok 😄 Let me replace it with boring:
```swift
guard let dto = ... else { return }
session.delete(dto)
```

Reply:
Your future yourself will thank me!

Reply:
btw, why do we even have `try` there? Under which circumstances the function will throw?

Reply:
It throws the error if there's no `MessageDTO` in the database

Reply:
updated in a4ea692b67bf9ebbfe08525503d4459ef18b14ac